src/demo/renderer.js

```diff
@@ -498,9 +504,75 @@ const drawPredictBot = state => {
   ctx.drawImage(botImg, botX, botY);
 };
 
+const drawWordFishImages = () => {
+  const canvas = getState().canvas;
+  const ctx = canvas.getContext('2d');
+  const state = getState();
+
+  const fishScale = 0.7;
+
+  const t = $time();
+  state.wordFish.forEach(fish => {
+    const swayValue = ((t * 360) / (20 * 1000)) % 360;
+    const swayOffsetY = Math.sin(((swayValue * Math.PI) / 180) * 3) / 20;
+
+    const xy = fish.getXY();
+    if (!fish.startTime) {
+      fish.startTime = t;
+      fish.speed = randomInt(10000, 15000);
+    }
+    let finalX;
+    if (fish.faceLeft) {
+      finalX =
+        constants.canvasWidth +
+        constants.fishCanvasWidth -
+        (constants.canvasWidth / fish.speed) * (t - fish.startTime);
+    } else {
+      finalX =
+        (constants.canvasWidth / fish.speed) * (t - fish.startTime) -
+        constants.fishCanvasWidth;
+    }
+    const finalY = xy.y + swayOffsetY;
+    fish.setXY({x: finalX, y: finalY});
+
+    drawSingleFish(fish, finalX, finalY, ctx, fishScale);
+  });
+
+  let wordFish = state.wordFish;
+  wordFish = wordFish.filter(
+    f =>
+      f.xy.x <= (constants.canvasWidth + constants.fishCanvasWidth) &&
+      f.xy.x >= -constants.fishCanvasWidth
+  );
+  const lastFish = wordFish[wordFish.length - 1];
+
+  if (
+    wordFish.length <= 1 ||
+    (wordFish.length <= 10 &&
+      lastFish.startTime > t &&
+      randomInt(0, wordFish.length * 500) <= 0)
+  ) {
+    const possibleFishComponents = filterFishComponents(
+    fishData,
+    state.appMode
+  );
+    const newFish = new FishOceanObject(state.fishCount,possibleFishComponents);
+    newFish.randomize();
+    const lane = randomInt(
+      0,
+      Math.floor(constants.canvasHeight / (fishScale * constants.fishCanvasHeight)) - 1
+    );
+    const y = lane * constants.fishCanvasHeight * fishScale;
+    newFish.setXY({x: -constants.fishCanvasWidth * 1.5, y});
+    newFish.faceLeft = lane % 2 === 0 ? true : false;
+    wordFish.push(newFish);
+    setState({fishCount: state.fishCount + 1});
+  }
+  setState({wordFish});
```

Comment:
Is it worth clearing `wordFish` from the state at the end of the scene, rather than just at its `init`, to reduce memory usage?

Reply:
Do we clear any of the fish lists?